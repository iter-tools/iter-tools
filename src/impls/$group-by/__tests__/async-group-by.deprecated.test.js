/**
 * @generated-from ./$group-by.deprecated.test.js
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

import { asyncGroupBy } from 'iter-tools-es';
import { asyncUnwrapDeep } from '../../../test/async-helpers.js';

describe('asyncGroupBy (deprecated)', () => {
  it('returns source values grouped by identity', async () => {
    const source = 'ABBAAC';
    const result = [
      ['A', ['A']],
      ['B', ['B', 'B']],
      ['A', ['A', 'A']],
      ['C', ['C']],
    ];
    expect(await asyncUnwrapDeep(asyncGroupBy(null, source))).toEqual(result);
    expect(await asyncUnwrapDeep(asyncGroupBy(null, source))).toEqual(result);
    expect(console.warn).callsMatchSnapshot();
  });
});
