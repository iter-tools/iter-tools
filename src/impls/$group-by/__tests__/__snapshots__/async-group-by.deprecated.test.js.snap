// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asyncGroupBy (deprecated) returns source values grouped by identity 1`] = `
Array [
  Array [
    "\`asyncGroupBy(null, iterable)\` is deprecated and will be removed in iter-tools@8. Instead use asyncGroup(iterable)",
  ],
]
`;
